core/storage/filestore.go

```diff
@@ -0,0 +1,30 @@
+package storage
+
+// Filestore is a interface for file storage
+type Filestore interface {
+	Get(StoredDataType, string) (string, error)
+	List(StoredDataType, string) ([]string, error)
+	Put(StoredDataType, string, string) error
+}
+
+// StoredDataType designates which type of data is stored
+type StoredDataType string
+
+const (
+	// SGFInput storage directory
```

Comment:
These names seem a little too wordy to me. I've provided some suggestions to abbreviate.